src/controllers/api/accreditation.ts

```diff
@@ -0,0 +1,363 @@
+import type { Request, Response } from 'express';
+import type { VerifiableCredential } from '@veramo/core';
+import type { DIDAccreditationRequestBody, DIDAccreditationRequestParams } from '../../types/accreditation.js';
+import type { ICredentialTrack, ITrackOperation } from '../../types/track.js';
+import type { CredentialRequest } from '../../types/credential.js';
+import { StatusCodes } from 'http-status-codes';
+import { v4 } from 'uuid';
+import { AccreditationRequestType, DIDAccreditationTypes } from '../../types/accreditation.js';
+import { CredentialConnectors, VerifyCredentialRequestQuery } from '../../types/credential.js';
+import { OperationCategoryNameEnum, OperationNameEnum } from '../../types/constants.js';
+import { IdentityServiceStrategySetup } from '../../services/identity/index.js';
+import { AccreditationService } from '../../services/api/accreditation.js';
+import { Credentials } from '../../services/api/credentials.js';
+import { eventTracker } from '../../services/track/tracker.js';
+import { body, query } from '../validator/index.js';
+import { validate } from '../validator/decorator.js';
+
+export class AccreditationController {
+	public static issueValidator = [
+		query('accreditationType')
+			.exists()
+			.isIn([
+				AccreditationRequestType.authorize,
+				AccreditationRequestType.accredit,
+				AccreditationRequestType.attest,
+			])
+			.bail(),
+		body('issuerDid').exists().isString().isDID().bail(),
+		body('subjectDid').exists().isString().isDID().bail(),
+		body('schemas').exists().isArray().bail(),
+		body('schemas.*.url').isURL().bail(),
+		body('schemas.*.type').custom(
+			(value) => typeof value === 'string' || (Array.isArray(value) && typeof value[0] === 'string')
+		),
+		body('parentAccreditation').optional().bail(),
+		body('rootAuthorization').optional().bail(),
+		query('accreditationType')
+			.custom((value, { req }) => {
+				const { parentAccreditation, rootAuthorization } = req.body;
+
+				const hasParentOrRoot = parentAccreditation || rootAuthorization;
+
+				if (
+					!hasParentOrRoot &&
+					(value === AccreditationRequestType.accredit || value === AccreditationRequestType.attest)
+				) {
+					throw new Error('parentAccreditation or rootAuthorization is required');
+				}
+
+				if (hasParentOrRoot && value === AccreditationRequestType.authorize) {
+					throw new Error(
+						'parentAccreditation or rootAuthorization is not required for an authorize operation'
+					);
+				}
```

Comment:
@Tweeddalex can u clarify this. Is parentAccreditation or rootAuthroization required for a root DID

Reply:
Hey @DaevMithran yeah I think it should only be trustFramework or trustFrameworkId if its a root authorization. But this might be something we tweak later as we start testing actual trust registries